joomla/joomla-cms
# Allow including of "parent" menu manifests

When we have multiple layouts (menu item types) for the same component view, we more or less copy the whole manifest and just add or remove a few parameters from it. The rest is duplicated code.
This can be seen for example when looking at the manifests for the com_content category view and its two layouts ([blog](https://github.com/joomla/joomla-cms/blob/staging/components/com_content/views/category/tmpl/blog.xml) and [list/default](https://github.com/joomla/joomla-cms/blob/staging/components/com_content/views/category/tmpl/default.xml)).
When we add a new feature to the view itself, we have to maintain the parameters in both layout manifests.
#### Summary of Changes

This PR makes use of the existing (probably not very well known) view and component manifest feature. Instead of duplicating the parameters in each layout manifest, we now can just specify to include the view and/or component manifest file and put the shared parameters into those manifests.
With this PR Joomla will now first load the layout manifest file (eg the default.xml one) and if it specifies to load one of the "parent" files it will load the view and/or the component one (metadata.xml).

When parameters are specified in multiple files, then the layout file takes precedence over the view file over the component file.
Also parameters specified in the "parent" files will appear _after_ the parameters defined in the layout file.

This PR changes the menu item model to load those option files and moves the shared parameters from the com_content category view to the view manfiest file so we have a good test case :smile: 
#### Testing Instructions
- Check the parameters in the category view of com_content (`category list` and `category blog`) and verify that all are still there after applying the PR
- Check the functionality of the parameters, there should be no change at all.
#### Side Effects

When working on this PR I detected a few inconsistencies between the two layouts:
- The blog layout has an additional option for the parameters in the "Options" tab called "Use Article Settings". That option was missing in the list layout. It's now available in both layouts.
- The blog layout has two additional parameters in the "Options" tab called `Position of Article Info` and `Show Tags`. Those are used in the blog layout itself (and not in the list layout), however in the single article view they exist as well. We could move them to the view file as well but I left it for now since it is a good showcase for how it will work :smile:
